opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -113,11 +154,31 @@ class SpanContext:
     """
 
     def __init__(self,
-                 trace_id: str,
-                 span_id: str,
+                 trace_id: int,
+                 span_id: int,
                  options: 'TraceOptions',
                  state: 'TraceState') -> None:
-        pass
+        self.trace_id = trace_id
+        self.span_id = span_id
+        self.options = options
+        self.state = state
+
+    def is_valid(self) -> bool:
+        """Get whether this `SpanContext` is valid.
+
+        A `SpanContext` is said to be invalid if its trace ID and span ID are
```

Comment:
Should this be "both" or "either"? https://www.w3.org/TR/trace-context/#field-value

Reply:
I was comparing this to the static invalid span which has both all-0 trace and span IDs, but I think you're right that this should be either.